### PR TITLE
Use pre-commit

### PR DESCRIPTION
**Pull request by wismill**
See [this comment](https://github.com/xkbcommon/libxkbcommon/issues/376#issuecomment-1738227651) in #376.

TODO:
- [ ] C linter/formatter: there is an old config file for [uncrustify](https://uncrustify.sourceforge.net/) in the repo.
  
  this will lead to a potentially massive diff.

Note that this PR currently only deal with _local_ pre-commit.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+- repo: https://github.com/psf/black
+  rev: 23.9.1
+  hooks:
+  - id: black
+    args: ['--check', '--diff', '.']
+- repo: https://github.com/charliermarsh/ruff-pre-commit
+  rev: v0.0.291
+  hooks:
+  - id: ruff
+    args: ['--ignore=E741,E501', '.']
```
Comment (whot) [inline .pre-commit-config.yaml:20]:
```suggestion
    # ambiguous-variable-name (E741), line-too-long (E501)
    args: ['--ignore=E741,E501', '.']
```

**Review by whot — APPROVED** [review]
Minor nitpick but otherwise this LGTM

File: .pre-commit-config.yaml (hunk 1 above)
Comment (wismill) [reply .pre-commit-config.yaml:21]: Good idea

**Comment by wismill** [comment]
I leave the C linter/formatter for a follow-up, as this can have a considerable impact on the code base. I think we should enforce the current style atm.

**Pull request merged**